src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs

```diff
@@ -113,10 +114,21 @@ public void AddDirectory (string folder, string folderInArchive, CompressionMeth
 		/// <summary>
 		/// HACK: aapt2 is creating zip entries on Windows such as `assets\subfolder/asset2.txt`
 		/// </summary>
-		public void FixupWindowsPathSeparators (Action<string, string> onRename)
+		public void FixupWindowsPathSeparators (HashSet<ulong> deletedEntries, Action<string, string> onRename)
 		{
 			bool modified = false;
-			foreach (var entry in zip) {
+
+			long entryCount = zip.EntryCount;
+			if (entryCount < 0)
+				return;
+
+			for (ulong i = 0; i < (ulong) entryCount; ++i) {
+				// If an entry index has been deleted then ReadEntry will throw a "Entry has been deleted" ZipException.
+				// So we skip deleted entries instead
+				if (deletedEntries.Contains (i))
+					continue;
+
+				var entry = zip.ReadEntry (i);
```

Comment:
`ZipEntry entry` has an `Index` property:

https://github.com/xamarin/LibZipSharp/blob/05c43d01f93a21e56294716faffca49c1deb0450/ZipEntry.cs#L61

Can we keep using the `foreach` loop and just `continue`, something like:

```suggestion
			foreach (var entry in zip) {
				if (deletedEntries.Contains (entry.Index))
					continue;
```

Reply:
Unfortunately, not. The foreach enumerator calls ReadEntry for each entry in the zip, which throws an exception if it's deleted. Ultimately, libZipSharp should probably be updated to handle that differently, but I didn't want to change it and enumerating by index is a bit more efficient anyway.